harpoon/suite_test.go

```diff
@@ -42,38 +43,33 @@ func TestSuite(t *testing.T) {
 	suite.RunT(t)
 }
 
-func stubGiven(ctx context.Context) error {
-	T(ctx).ApplyFixture(ctx, "stub")
-
-	return nil
+func stubGiven(ctx context.Context, t TestingT) {
+	t.ApplyFixture(ctx, "stub")
 }
 
-func stubWhen(ctx context.Context, key, value string) error {
-	t := T(ctx)
-
+func stubWhen(ctx context.Context, t TestingT, key, value string) {
 	var configMap corev1.ConfigMap
 	require.NoError(t, t.Get(ctx, t.ResourceKey("stub-config-map"), &configMap))
 
 	configMap.Data = map[string]string{key: value}
 
 	require.NoError(t, t.Update(ctx, &configMap))
-
-	return nil
 }
 
-func stubThen(ctx context.Context, key, value string) error {
-	t := T(ctx)
-
+func stubThen(ctx context.Context, t TestingT, key, value string) {
 	var configMap corev1.ConfigMap
 	require.NoError(t, t.Get(ctx, t.ResourceKey("stub-config-map"), &configMap))
 
 	require.Equal(t, value, configMap.Data[key])
+}
 
-	return nil
+func stubAnd(t TestingT) {
+	require.Equal(t, 1, 1)
 }
```

Comment:
Could you help me understand why `stubAnd` has any call? As far as I understand you would like to have last step that should be executed without any assertion, so that whole suite machinery will recover and catch any potential error. Am I right?

Reply:
This is more of just an example of the step definition signatures that I added to show passing `TestingT` with no context. Generally your last step is where you'll do all of the side effect assertions with the whole "Given" (i.e. setup steps), "When" (i.e. some action with side-effects), "Then" (i.e. assertions) format of Gherkin tests.

> so that whole suite machinery will recover and catch any potential error

With the changes I made basically what will happen is that if an assertion using something like `require` fails, i.e. `require.Equal(t, 1, 2)`, then rather than doing a panic and immediately halting the test, you'll just see a typical test failure and the rest of the tests in other scenarios will still run.